### PR TITLE
Feat: Add creator fields as filters and to list view

### PR DESCRIPTION
**Pull request by Feranchz**
This PR builds upon the progress made in https://github.com/strapi/strapi/pull/17043 solving a problem with the traverse functions on sanitize and adding test cases. 

### What does it do?

1. Two new filters, "Created by" and "Updated by," have been added to the list view in the content manager.

### How to test it?

- You should be able to select a "Created by" or "Updated by" filter and filter by them. The tag should contain the user display name
- You should be able to enter directly to a url with filters (example: /admin/content-manager/collectionType/api::address.address?page=1&pageSize=10&sort=postal_code:ASC&filters[$and][0][createdBy][id][$eq]=1) and see the tags with the display name of every user correctly

### CODE REVIEW

File: packages/core/admin/server/services/__tests__/permissions-manager-sanitize.test.js [hunk 1]
```diff
@@ -97,4 +97,21 @@ describe('Permissions Manager - Sanitize', () => {
       expect(result).toEqual({ c: 'Bar' });
     });
   });
+
+  describe('Sanitize Query', () => {
+    it('Removes hidden fields on filters, sort, populate and fields', async () => {
+      const data = {
+        filters: { a: 'Foo', c: 'Bar' },
+        sort: { a: 'asc', c: 'desc' },
+        populate: { a: 'Foo', c: 'Bar' },
```
Comment (Convly) [inline packages/core/admin/server/services/__tests__/permissions-manager-sanitize.test.js:106]:
```suggestion
        populate: { a: true, c: true },
```

**Review by joshuaellis — APPROVED** [review]
FE code looks fine to me, I've not QAd but it doesn't look like it's changed from it's predecessor.

**Review by Marc-Roig — APPROVED** [review]

**Comment by gu-stav** [comment]
@Marc-Roig Just do double-check: do we want to have this work in a patch-release or 4.13? I'm really not sure, but I think I would prefer 4.13.

**Comment by Marc-Roig** [comment]
I would prefer 4.13 too

**Review by Convly — APPROVED** [review]

**Comment by gu-stav** [comment]
@Feranchz Since you have merged it, this will land in 4.12.1, but since you have also added the 4.13 milestone I suppose this was a mistake?

**Comment by Marc-Roig** [comment]
yes :D better to revert and create a new pr

**Pull request merged**